ovos_plugin_manager/thirdparty/sr.py

```
@@ -0,0 +1,466 @@
+            - Converts big-endian input to little-endian on the fly.
+            - If `samples_24_bit_pretending_to_be_32_bit` is set, expands 24-bit samples into 32-bit little-endian samples.
+            - Converts multi-channel input to mono by mixing channels equally.
+            
+            Parameters:
+                size (int): Number of frames to read from the underlying reader; -1 means "read all available frames".
+            
+            Returns:
+                bytes: PCM audio data containing mono, little-endian samples (may be 32-bit if 24-bit-to-32-bit expansion occurred).
+            """
+            buffer = self.audio_reader.readframes(self.audio_reader.getnframes() if size == -1 else size)
+            if not isinstance(buffer, bytes): buffer = b""  # workaround for https://bugs.python.org/issue24608
```

Ruff (0.14.14)

Multiple statements on one line (colon)

(E701)